src/panels/lovelace/components/hui-energy-period-selector.ts

```diff
@@ -37,41 +68,105 @@ export class HuiEnergyPeriodSelector extends SubscribeMixin(LitElement) {
     return html`
       <div class="row">
         <div class="label">
-          ${formatDate(this._startDate, this.hass.locale)}
+          ${this._period === "day"
+            ? formatDate(this._startDate, this.hass.locale)
+            : this._period === "month"
+            ? this._startDate.toLocaleString(this.hass.locale.language, {
+                month: "long",
+              })
+            : this._period === "year"
+            ? this._startDate.toLocaleString(this.hass.locale.language, {
+                year: "numeric",
+              })
+            : `${formatDateShort(
+                this._startDate,
+                this.hass.locale
+              )} - ${formatDateShort(
+                this._endDate || new Date(),
+                this.hass.locale
+              )}`}
         </div>
 
-        <mwc-icon-button label="Previous Day" @click=${this._pickPreviousDay}>
+        <mwc-icon-button label="Previous" @click=${this._pickPrevious}>
           <ha-svg-icon .path=${mdiChevronLeft}></ha-svg-icon>
         </mwc-icon-button>
-        <mwc-icon-button label="Next Day" @click=${this._pickNextDay}>
+        <mwc-icon-button label="Next" @click=${this._pickNext}>
           <ha-svg-icon .path=${mdiChevronRight}></ha-svg-icon>
         </mwc-icon-button>
 
-        <mwc-button
+        <mwc-button dense outlined @click=${this._pickToday}>
+          Today
+        </mwc-button>
+        <ha-button-toggle-group
+          .buttons=${viewButtons}
+          .active=${this._period}
           dense
-          outlined
-          .disabled=${isToday(this._startDate)}
-          @click=${this._pickToday}
-          >Today</mwc-button
-        >
+          @value-changed=${this._handleView}
+        ></ha-button-toggle-group>
       </div>
     `;
   }
 
+  private _handleView(ev: CustomEvent): void {
+    this._period = ev.detail.value;
+    const currentStart = this._startDate || startOfToday();
+    this._setDate(
+      this._period === "day"
+        ? currentStart
+        : this._period === "week"
+        ? startOfWeek(currentStart, { weekStartsOn: 1 })
+        : this._period === "month"
+        ? startOfMonth(currentStart)
+        : startOfYear(currentStart)
```

Comment:
I think that if we're scaling down (ie year to month), we should check if the current date falls in that range and pick that. Now when I click Year (2021) -> Month (January) -> Week (Dec 28-Jan 3) -> Year (2020)

Reply:
Yeah, had thought of that too, implemented it now, and think I like it :-)